src/mosaic-node-manager.cc

```diff
@@ -76,25 +114,71 @@ namespace ns3 {
         NS_LOG_INFO("Created node " << singleNode->GetId());
         m_mosaic2ns3ID[ID] = singleNode->GetId();
 
-        //Install Wave device
-        NS_LOG_INFO("Install WAVE on node " << singleNode->GetId());
-        InternetStackHelper internet;   
-        internet.Install(singleNode);
-        NetDeviceContainer netDevices = m_wifi80211pHelper.Install(m_wifiPhyHelper, m_waveMacHelper, singleNode);
-        m_ipAddressHelper.Assign(netDevices);
-
-        //Install app
-        NS_LOG_INFO("Install MosaicProxyApp application on node " << singleNode->GetId());
-        Ptr<MosaicProxyApp> app = CreateObject<MosaicProxyApp>();
-        app->SetNodeManager(this);
-        singleNode->AddApplication(app);
-        app->SetSockets();
-
-        //Install mobility model
-        NS_LOG_INFO("Install MosaicMobilityModel on node " << singleNode->GetId());
-        Ptr<ConstantVelocityMobilityModel> mobModel = CreateObject<ConstantVelocityMobilityModel>();
-        mobModel->SetPosition(position);
-        singleNode->AggregateObject(mobModel);
+        // Install the appropriate device based on communication type
+        if (commType == DSRC) {
+            NS_LOG_INFO ("Creating wifi helpers for the node...");
+            InternetStackHelper internet;   
+            internet.Install(singleNode);
+            NetDeviceContainer netDevices = m_wifi80211pHelper.Install(m_wifiPhyHelper, m_waveMacHelper, singleNode);
+            m_ipAddressHelper.Assign(netDevices);
+
+            //Install app
+            NS_LOG_INFO("Install MosaicProxyApp application on node " << singleNode->GetId());
+            Ptr<MosaicProxyApp> app = CreateObject<MosaicProxyApp>();
+            app->SetNodeManager(this);
+            singleNode->AddApplication(app);
+            app->SetSockets();
+
+            //Install mobility model
+            NS_LOG_INFO("Install MosaicMobilityModel on node " << singleNode->GetId());
```

Comment:
Not sure if this is how it was in the original code, but the log message says "MosaicMobilityModel" but the actual model is a ConstantVelocityMobilityModel. just want to check that this is intentional.